libraries/chain/db_maint.cpp

```diff
@@ -167,15 +168,21 @@ void database::pay_workers( share_type& budget )
       return wa.id < wb.id;
    });
 
+   const auto last_budget_time = get_dynamic_global_properties().last_budget_time;
+   const auto passed_time_ms = head_time - last_budget_time;
+   const bool passed_time_is_a_day = ( passed_time_ms == fc::days(1) );
+   // the variable above is more likely false on BitShares mainnet, so do calculations below anyway
```

Comment:
IMO can simplify the code and remove the passed_time_is_a_day check because it's unlikely to ever be true